Source/stores.cpp

```diff
@@ -1890,43 +1867,43 @@ void StorytellerIdentifyItem()
 	} else {
 		myPlayer.InvList[idx]._iIdentified = true;
 	}
-	myPlayer.HoldItem._iIdentified = true;
-	TakePlrsMoney(myPlayer.HoldItem._iIvalue);
+	item._iIdentified = true;
+	TakePlrsMoney(item._iIvalue);
 	CalcPlrInv(myPlayer, true);
 }
 
-void ConfirmEnter()
+void ConfirmEnter(Item &item)
 {
 	if (stextsel == 18) {
 		switch (stextshold) {
 		case STORE_SBUY:
-			SmithBuyItem();
+			SmithBuyItem(item);
 			break;
 		case STORE_SSELL:
 		case STORE_WSELL:
 			StoreSellItem();
 			break;
 		case STORE_SREPAIR:
-			SmithRepairItem();
+			SmithRepairItem(item._iIvalue);
 			break;
 		case STORE_WBUY:
-			WitchBuyItem();
+			WitchBuyItem(item);
 			break;
 		case STORE_WRECHARGE:
-			WitchRechargeItem();
+			WitchRechargeItem(item._iIvalue);
 			break;
 		case STORE_BBOY:
-			BoyBuyItem();
+			BoyBuyItem(item);
 			break;
 		case STORE_HBUY:
-			HealerBuyItem();
+			HealerBuyItem(item);
 			break;
 		case STORE_SIDENTIFY:
-			StorytellerIdentifyItem();
+			StorytellerIdentifyItem(item);
 			StartStore(STORE_IDSHOW);

```

Comment:
StorytellerIdentifyItem doesn't really need the item as an argument, it's more so StartStore(STORE_IDSHOW) knows which item was identified to display to the player. It could take the selected index or cost.

Reply:
If I only give it the price i have to set it as identified directly here, or in `StartStorytellerIdentifyShow()` which seams like the wrong place to do so. Keeping it like this feels like it keeps it clear what to refactor with out things getting more misleading then they where.